tests/integration/rlimits.bats

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	# Do not change the Cur value to be equal to the Max value
+	# Because in some environments, the soft and hard nofile limit have the same value.
+	[ $EUID -eq 0 ] && prlimit --nofile=1024:65536 -p $$
+	setup_busybox
+}
+
+function teardown() {
+	teardown_bundle
+}
+
+@test "runc run with RLIMIT_NOFILE(The same as system's hard value)" {
+	# https://github.com/opencontainers/runc/pull/4265#discussion_r1588599809
+	hard=$(ulimit -n -H)
+	soft="$hard"
+	update_config ".process.rlimits = [{\"type\": \"RLIMIT_NOFILE\", \"hard\": ${hard}, \"soft\": ${soft}}]"
+	update_config '.process.args = ["/bin/sh", "-c", "ulimit -n"]'
+
+	runc run test_ulimit
+	[ "$status" -eq 0 ]
+	[[ "${output}" == "${soft}" ]]
+}
+
+@test "runc run with RLIMIT_NOFILE(Bigger than system's hard value)" {
+	requires root
+	# https://github.com/opencontainers/runc/pull/4265#discussion_r1588599809
+	hard=$(ulimit -n -H)
+	soft=$((hard + 1))
```

Comment:
Maybe something like
```bash
lim=$(ulimit -n -H)
hard=$((lim+1))
soft="$hard"
update_config ".process.rlimits = [{\"type\": \"RLIMIT_NOFILE\", \"soft\": ${soft}, \"hard\": ${hard}}]"
```
So it's clear what we set.

Also, we can move this code into a separate function to avoid repetition.

E.g.

```bash
# Sets and check rlimit_nofile for runc run. Arguments are:
#  $1: soft limit;
#  $2: hard limit.
function run_check_nofile() {
	soft="$1"
	hard="$2"
	update_config ".process.rlimits = [{\"type\": \"RLIMIT_NOFILE\", \"sioft\": ${soft}, \"hard\": ${hard}}]"
	update_config '.process.args = ["/bin/sh", "-c", "ulimit -n; ulimit -H -n"]'
	
	runc run test_rlimit
	[ "$status" -eq 0 ]
	[[ "${lines[0]}" == "${soft}" ]]
	[[ "${lines[1]}" == "${hard}" ]]
}

# Sets and check rlimit_nofile for runc exec. Arguments are:
#  $1: soft limit;
#  $2: hard limit.
function exec_check_nofile() {
	soft="$1"
	hard="$2"
	update_config ".process.rlimits = [{\"type\": \"RLIMIT_NOFILE\", \"sioft\": ${soft}, \"hard\": ${hard}}]"
	
	runc run test_rlimit
	[ "$status" -eq 0 ]
	
	runc exec test_busybox /bin/sh -c "ulimit -n; ulimit -H -n"
	[ "$status" -eq 0 ]
	[[ "${lines[0]}" == "${soft}" ]]
	[[ "${lines[1]}" == "${hard}" ]]
}
```

and call it from all tests:
```bash
val=$(ulimit -n -H)
run_check_nofile "$val" "$val"
```